akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -522,8 +522,9 @@ final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
           pull(in)
         } catch {
           case NonFatal(ex) ⇒ decider(ex) match {
-            case Supervision.Stop ⇒ failStage(ex)
-            case _ ⇒
+            case Supervision.Stop   ⇒ failStage(ex)
```

Comment:
Ah, misunderstood you twice in three minutes. Sorry again. So the reason would be that _if_ the stage is faulty it will apply the supervision strategy which is weird because it can likely never recover from that. Right?